shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx

```diff
@@ -0,0 +1,40 @@
+<PlatformWrapper platform="react-native">
+1.  **Add the required variables**
+
+    In `App.tsx`, add the following to the declarations:
+
+    ```typescript
+    // In a production environment, you retrieve the key and salt from
+    // an authentication server. For this code example you generate locally.
+    const encryptionKey = '';
+    const encryptionSaltBase64 = '';
+    ```
+
+2.  **Import the required modules**
+
+    In `App.tsx`, add the following import in `import {} from 'react-native-agora-rtc-ng';`:
+
+    ```typescript
+    EncryptionMode,
+    ```
+
+3.  **Call the channel encryption method to enable channel encryption**
+
+    To enable channel encryption in your <Vpl k="CLIENT" />, you need to:
+
+    1. Create an object of `EncryptionConfig` class and specify a configuration for the channel encryption. In the configuration, you specify `encryptionMode`, `encryptionKey`, and `encryptionKdfSalt`. `encryptionKdfSalt` is required for `Aes128Gcm2` and `Aes256Gcm2` modes but optional otherwise.
+
+    2. Call `enableEncryption` and pass `true` and the `EncryptionConfig` object as parameters to enable media encryption.
+
+    To implement this logic, in `App.tsx`, add the following code after `agoraEngine.initialize({appId: appID});`
+
+    ```typescript
+    let encryptionConfig = {
+        encryptionBase64: encryptionBase64,
+        encryptionKey: encryptionKey,
+        encryptionMode: EncryptionMode.Aes128Ecb,
+      };
+    engine.enableEncryption(true, encryptionConfig);
+    ```
+    <Vg k="COMPANY"/> recommends using `Aes128Gcm2` or `Aes256Gcm2` encrypted mode. These two modes support the use of salt for higher security.
```

Comment:
Please don't:
* Say Agora recommends, always "best practice"
* Put the number of items you are about to talk about. Chances are you will add or remove an item and forget about the number. Keep it simple. 

Best practice is to use  `Aes128Gcm2` or `Aes256Gcm2`. These modes use salt for higher security.